examples/with-logger/main/helpers/Logger.ts

```diff
@@ -0,0 +1,184 @@
+const path = require("path");
+const fs = require("fs");
+import test from "../../package.json";
+import { ipcMain } from "electron";
+import { app } from "electron";
+
+export const LOGGER_CODES = {
+  DEBUG: 0,
+  INFO: 1,
+  WARNING: 2,
+  ERROR: 3,
+  FATAL: 4,
+};
+
+export const LOGGER_SOURCE = {
+  BACK: 0,
+  FRONT: 1,
+  LOGGER: 2,
+};
+
+export function getNameFromCode(code, enumeration) {
+  for (let prop in enumeration) {
+    if (enumeration[prop] === code) {
+      return prop;
+    }
+  }
+}
+
+export class Logger {
+  static _instance: Logger;
+  isDev: boolean;
+  todayCode: string;
+  logFileName: string;
+  fileExtension: any;
+  fileIdentifier: string;
+  logFolder: any;
+  numberOfLogFilesAccepted: any;
+  static build(opts) {
+    if (!Logger._instance) Logger._instance = new Logger(opts);
+  }
+
+  static get() {
+    if (!Logger._instance) throw "Error, logger not instanciated !";
+    return Logger._instance;
+  }
+
+  constructor(opts) {
+    this.isDev = opts.isDev || false;
+    var currentDate = new Date();
+    var dd = String(currentDate.getDate()).padStart(2, "0");
+    var mm = String(currentDate.getMonth() + 1).padStart(2, "0"); //January is 0!
+    var yyyy = currentDate.getFullYear();
```

Comment:
If variables are unchanged after they are defined please use 'const' instead of 'var' if not use 'let' instead.

```suggestion
    const currentDate = new Date();
    const dd = String(currentDate.getDate()).padStart(2, "0");
    const mm = String(currentDate.getMonth() + 1).padStart(2, "0"); //January is 0!
    const yyyy = currentDate.getFullYear();
```